backend/routes/appointments/viewAppointmentsRouter.js

```diff
@@ -51,10 +51,52 @@ module.exports = ({ prisonApi, whereaboutsApi, logError }) => async (req, res) =
       court: appointment.court,
     }))
 
+  const addBookingIdAndHearingTypesToVLBAppointments = appointment => {
+    const appCopy = { ...appointment }
+    if (appointment.appointmentTypeCode === 'VLB') {
+      videoLinkAppointments.forEach(vlAppointment => {
+        if (vlAppointment.appointmentId === appointment.id) {
+          appCopy.hearingType = vlAppointment.hearingType
+          appCopy.bookingId = vlAppointment.bookingId
+        }
+      })
+    }
+    return appCopy
+  }
+
+  const addDisplayTimesForVLBAppointments = (appointment, index, appointmentsArray) => {
+    const appCopy = { ...appointment }
+    const preAppointment = appointmentsArray.find(
+      otherAppointment => otherAppointment.bookingId === appointment.bookingId && otherAppointment.hearingType === 'PRE'
+    )
+    const postAppointment = appointmentsArray.find(
+      otherAppointment =>
+        otherAppointment.bookingId === appointment.bookingId && otherAppointment.hearingType === 'POST'
+    )
+    if (preAppointment) {
+      appCopy.displayStartTime = preAppointment.startTime
+    }
+    if (postAppointment) {
+      appCopy.displayEndTime = postAppointment.endTime
+    }
+    return appCopy
+  }
+
   const appointmentsEnhanced = appointments
     .filter(appointment => (type ? appointment.appointmentTypeCode === type : true))
+    .map(appointment => addBookingIdAndHearingTypesToVLBAppointments(appointment))
```

Comment:
This is good work - there is a nice separation of concerns. Probably better than what i would have done (which its put it all in 1 function)